src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleById } from "../utils/api";
+import { MessageSquare, Clock } from "react-feather";
+
+function SingleArticle() {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState({});
+  const [articleCreatedAt, setArticleCreatedAt] = useState("");
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getArticleById(article_id)
+      .then((res) => {
+        setIsLoading(false);
+        setArticle(res.data.article);
+        const parsedDate = new Date(res.data.article.created_at);
+        setArticleCreatedAt(parsedDate.toDateString());
```

Comment:
This doesn't need new state, while you cannot mutate the article state, you could absolutely make a pure function that maps over article and updates the date. Or you could even use a library. Day.js is good for this.